evalml/pipelines/component_graph.py

```diff
@@ -0,0 +1,213 @@
+        return self
+
+    def add_edge(self, from_component, to_component):
+        """Add an edge connecting two nodes of the component graph. Note that both nodes
+        must already exist inside the graph.
+
+        Arguments:
+            from_component (str or int): The parent node, to be computed first
+            to_component (str or int): The child node, to be computed after the parent
+        """
+        if from_component not in self.component_names.keys() or to_component not in self.component_names.keys():
+            raise ValueError("Cannot add an edge for a component not in the graph yet")
+        self._component_graph.add_edge(from_component, to_component)
+        self._recompute_order()
+        return self
+
+    def merge_graph(self, other_graph):
+        """Add all components and edges from another `ComponentGraph` object to this one.
+        Components with unique names will be added as unique nodes, even if the component object
+        is the same type as another. Components with identical names will be considered the
+        same node, and the object from the incoming graph will be saved as that component.
+
+        Arguments:
+            other_graph (ComponentGraph): The other graph to combine with this graph
+        """
+        for component_name, component in other_graph.component_names.items():
+            self.component_names[component_name] = component
+        self._component_graph.add_edges_from(other_graph._component_graph.edges)
+        self._recompute_order()
+        return self
+
+    def get_component(self, component_name):
+        """Retrieves a single component object from the graph.
+
+        Arguments:
+            component_name (str): Name of the component to retrieve
+
+        Returns:
+            ComponentBase object
+        """
+        try:
+            return self.component_names[component_name]
+        except KeyError:
+            raise ValueError(f'Component {component_name} is not in the graph')
+
+    def get_estimators(self):
+        """Gets a list of all the estimator components within this graph
+
+        Returns:
+            list: all estimator objects within the graph
+        """
+        estimators = []
+        for component in self.component_names.values():
+            if component.model_family is not ModelFamily.NONE:
+                estimators.append(component)
+        return estimators
+
+    def parents(self, component_name):
+        """Finds the names of all parent nodes of the given component
+
+        Arguments:
+            component_name (str): Name of the child component to look up
+
+        Returns:
+            iterator of parent component names
+        """
+        if component_name not in self.component_names.keys():
+            raise ValueError(f'Component {component_name} is not in the graph')
+        return self._component_graph.predecessors(component_name)
+
+    def graph(self, name, graph_format):
```

Comment:
This function is intended to be called by the `PipelineBase.graph()` function, as there are a few filepath handling things there that I've left to live within the pipeline function.